EIPS/eip-3074.md

```diff
@@ -27,125 +27,103 @@ While it is possible to emulate sponsored transactions (ex. [Gas Station Network
 
 ## Specification
 
-An opcode, at `0xf9`, must function like a `CALL` (`0xF1`) instruction:
- - To the `to` address,
- - Transferring `value` wei from the invoker to the callee,
- - Passing `gas` gas for execution,
- - With calldata in the memory region specified by `argsOffset` and `argsLength`,
- - With a return data region specified by `retOffset` and `retLength`.
+### Conventions
 
-Additionally, the opcode must:
+ - **`top - N`** - the `N`th most recently pushed value on the EVM stack, where `top - 0` is the most recent.
+ - **`||`** - byte concatenation operator.
 
- - Set the caller address based on an ECDSA signature.
+### Constants
 
-`CALLFROM` must increase the call depth by one, in the same way as `CALL`. `CALLFROM` must not increase the call depth by two (as it would if it first called into the sponsee account and then into the callee.)
+|     Constant     | Value  |
+| ---------------- | ------ |
+| `TYPE`           | `0x03` |
 
-In a static context (such as the one created by `STATICCALL`), `CALLFROM` with a non-zero `value` must exit the current execution frame immediately (in the same way `CALL` behaves with a non-zero value in a static context.)
+`TYPE` is an [EIP-2718](./eip-2718.md) transaction type reserved for EIP-3074 signatures to prevent signature collisions.
 
-### Definitions
+### Context Variables
 
- - **`CALLFROM`** - the specific instruction encoded as `0xf9`, introduced by this EIP, which implements the `CALL` analogue.
- - **Transaction-like Package (TLP)** - the signed arguments passed to `CALLFROM`.
- - **Sponsor** - the account which is responsible for paying gas fees and sending the transaction.
- - **Sponsee** - the account which signed the transaction-like package.
- - **Invoker** - the account or contract which contains `CALLFROM`.
- - **Callee** - the target of the call from `CALLFROM`.
+|     Variable        | Type      | Initial Value                                                                        |
+| ------------------- | --------- |:------------------------------------------------------------------------------------ |
+| `authorizedAccount` | `address` | `0x0000000000000000000000000000000000000000` |
```

Comment:
Does `0x0000000000000000000000000000000000000000 == 0000000000000000000000000000000000000000000000000000000000000000`? Because we wouldn't want a change in address format to break things.

Reply:
I think one alternative would be to have it be an optional address (or address pointer), so that we don't have to use the zero address to indicate no authorized account

Reply:
Once pushed on the stack, addresses are extended to 32 bytes, right?

Reply:
@SamWilsn yes, I'm just trying to think if there is any weirdness that could occur.